src/main/scala/io/iohk/ethereum/ledger/Ledger.scala

```diff
@@ -33,7 +30,7 @@ class LedgerImpl(vm: VM, blockchainConfig: BlockchainConfig) extends Ledger with
     val blockExecResult = for {
       _ <- validateBlockBeforeExecution(block, blockchain, validators)
 
-      execResult <- executeBlockTransactions(block, blockchain, storages, validators)
+      execResult <- executeBlockTransactions(block, blockchain, storages, validators.signedTransactionValidator)
```

Comment:
You are right @LukasGasior1. The only question is when we do it. I'm happy to leave the shape as is until we get through the txs. I *think* that's the most efficient.